### ISSUE TITLE
ability to choose prefered port location when creating gcp connection

### ISSUE THREAD

**Issue by mi1os** [megaport/terraform-provider-megaport#4]
I have tried to attach MCR in Equinix PA2/3 to GCP europe-west3 (frankfurt). tf code looks like

```
resource megaport_gcp_connection sbx_par_1_gcp_fra_1 {
  vxc_name = "sbx-par-1-gcp-fra-1"
  rate_limit = 200

  csp_settings {
    attached_to = megaport_mcr.sbx_par_1.id
    pairing_key = google_compute_interconnect_attachment.kiwi["fra-ovh-par-1-a"].pairing_key
  }
}
```

it all worked just fine and VXC is up and running but the problem is that the B end port was chosen in Equinix LD5 which is not ideal path and would just add unnecessary latency. in UI when pairing key is entered we have possibility to choose from a list of available ports and I tend to choose the one closest to the MCR or destination region so I would choose either Paris or Frankurt. 

However the provider chooses either random one from the ports or the first one from randomly sorted list (altought I have tried it twice and got London both times). Having either possibility to set preferred port location or adding some logic to the selection where the provider would prefer ports from location of either MCR or the partner region(if at all possible) would be great.

**Comment by HammoTime** [comment]
Thanks for lodging, mi1os.

I'll look into this one and get back to you with an estimate in the next few weeks on when we will be able to get this in.

**Comment by arctiqhart** [comment]
Also experiencing this - VXC b_end is in New York (lga-zone1-16) and a_end are all in Canada. Need to be able to define the b_end within Terraform.

**Comment by kdw174** [comment]
I've created a pr for this issue and the required updates in megaportgo. I've tested creating a megaport_gcp_connection specifying the new requested_product_id field and confirmed it used the Google port in the defined location. I also tested omitting the new parameter I added and confirmed the default behavior of selecting the first matching location in the list. This change is blocked by the megaportgo pr: https://github.com/megaport/megaportgo/pull/2
